01-data-structures/06-trees/binary_tree/binary_search_tree.rb

```diff
@@ -3,19 +3,117 @@
 class BinarySearchTree
 
   def initialize(root)
+    @root = root
   end
 
   def insert(root, node)
+    last = nil
+    current = root
+
+    loop do 
+
+      if current.left != nil && current.rating >= node.rating 
+        current = current.left 
+      end
+
+      if current.right != nil && current.rating <= node.rating  
+        current = current.right
+      end
+
+      # if current.left == nil || current.right == nil 
+      #   if current.rating > node.rating
+      #     current.left = node
+      #     node.parent = current
+      #     break
+      #   elsif current.rating < node.rating
+      #     current.right = node
+      #     node.parent = current
+      #     break
+      #   end
+      # end
+
+      if current.left == nil 
+        if current.rating > node.rating 
+          current.left = node
+          node.parent = current  
+          break
+        end
+      end
+
+      if current.right == nil 
+        if current.rating < node.rating 
+          current.right = node 
+          node.parent = current
+          break 
+        end
+      end
+
+    end
   end
 
   # Recursive Depth First Search
   def find(root, data)
```

Comment:
- [ ] Your implementation is not DFS (Depth First Search), it's BFS (Breadth First Search) but that's OK. We can review the differences in our meeting if you'd like.